client/my-sites/post-type-list/post-actions-ellipsis-menu/copy-link.jsx

```diff
@@ -0,0 +1,64 @@
+/** @format */
+
+/**
+ * External dependencies
+ */
+import React from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import { localize } from 'i18n-calypso';
+
+/**
+ * Internal dependencies
+ */
+import PopoverMenuItemClipboard from 'components/popover/menu-item-clipboard';
+import { getPost } from 'state/posts/selectors';
+import { bumpStat, recordTracksEvent } from 'state/analytics/actions';
+import { bumpStatGenerator } from './utils';
+import { infoNotice } from 'state/notices/actions';
+
+function PostActionsEllipsisMenuCopyLink( { onCopyLinkClick, copyLink, infoNotice, translate } ) {
+	const onCopy = () => {
+		onCopyLinkClick();
+		infoNotice( translate( 'Link copied to clipboard.' ), { duration: 3000 } );
+	};
+	return (
+		<PopoverMenuItemClipboard text={ copyLink } onCopy={ onCopy }>
+			{ translate( 'Copy Link' ) }
+		</PopoverMenuItemClipboard>
+	);
+}
+
+PostActionsEllipsisMenuCopyLink.propTypes = {
+	onCopyLinkClick: PropTypes.func,
+	copyLink: PropTypes.string,
+	translate: PropTypes.func,
+};
+
+const mapStateToProps = ( state, { globalId } ) => {
+	const post = getPost( state, globalId );
+	if ( ! post ) {
+		return {};
+	}
+
+	return {
+		copyLink: post.URL,
+	};
+};
+
+const mapDispatchToProps = { bumpStat, infoNotice, recordTracksEvent };
+
+const mergeProps = ( stateProps, dispatchProps, ownProps ) => {
+	const bumpCopyLinkStat = bumpStatGenerator( stateProps.type, 'copyLink', dispatchProps.bumpStat );
```

Comment:
`type` is not in the state. Did you forget to include it on `mapStateToProps`?

```js
const mapStateToProps = ( state, { globalId } ) => {
	const post = getPost( state, globalId );
	if ( ! post ) {
		return {};
	}

	return {
		copyLink: post.URL,
		type: post.type,
	};
};
```

Reply:
Yep! 45b0b1df1a8f38ef8f543dc86f2bdb5507ef3fbe